src/pages/SongPage.tsx

```diff
@@ -26,8 +28,19 @@ const SongPage: React.FC = () => {
 
   useEffect(() => {
     getNumSongsForBookId(bookId).then((size) => setSongBookLength(size));
+
+    window.addEventListener("resize", ToggleNavButtonListeners);
+    return () => {
+      window.removeEventListener("resize", ToggleNavButtonListeners);
+    };
   }, [bookId]);
 
+  useEffect(() => {
+    // want to give it enough time before running the button hiding/showing function
+    // if we run it too soon, the buttons won't be in the dom yet
+    setTimeout(ToggleNavButtonListeners, 1000);
+  }, [songViewMode]);
```

Comment:
ok so this `setTimeout()` here feels a little dirty but I think it's necessary. The problem is, if your screen was already too narrow before getting to `songPage`, the buttons will stay rendered since the listener only listens to screen resize. So we have to run an initial check after everything is loaded. But if you run it without `setTimeout()`, it will run too fast and the buttons will not be rendered. Anything under 400 for the timeout was too fast for me. Since `setTimeout()` is kinda hacky, I tried the approach of waiting for the buttons to load in with [MutationObserver](https://stackoverflow.com/questions/5525071/how-to-wait-until-an-element-exists). But I couldn't access the style of the inserted dom node which I need to hide/display it. Maybe the style isn't loaded in when the dom node is inserted? Would also have to not run parts of the observer for the first/last page, since only 1 button can load on those two pages. I think its just easier to use a `setTimeout()`. Only problem is if someone's device is soooo slow that it renders the button after 1 second.

Reply:
Yeaaah I see how this is a little hacky, but I don't really know a better solution. We can revisit the UI later.